docs/v3/core/index.rst

```diff
@@ -1598,38 +1508,162 @@ Storage transformers may be stacked to combine different functionalities:
 
 .. _extensions_section:
 
-Extension points
-================
+Extensions
+==========
 
+This section describes how additional functionality can be defined
+for Zarr datasets by the `metadata documents`_.
+
+Extension points
+----------------
 
 Different types of extensions can exist and they can be grouped as follows:
 
-=========== ======================= ================================================
-level       extension               metadata
-=========== ======================= ================================================
-array       data type               `data_type`_
-array       chunk grid              `chunk_grid`_
-array       chunk key encoding      `chunk_key_encoding`_
-array       codecs                  `codecs`_
-array       storage transformer     `storage_transformers`_
-=========== ======================= ================================================
-
-If such extension points are used by groups or arrays, they are required.
-
-See https://github.com/zarr-developers/zarr-specs/issues/49 for a list of
-potential extensions.
-
-Specifications for new extensions are recommended to be published in the
-https://github.com/zarr-developers/zarr-specs repository via the
-`ZEP process <https://zarr.dev/zeps/active/ZEP0000.html>`_. If a specification
-is published decentralized (e.g. for initial experimentation or due to a very
-specialized scope), it must use a URL in the `name` key of its metadata, which
-identifies the publishing organization or individual, and should point to the
-specification of the extension.
-
-Future versions of this specification may also add new core features by adding new top-level
-metadata keys. Such features are required by default. However, if the value of an unknown feature
-is an object containing the key-value pair ``"must_understand": false``, it can be ignored.
+=========== ======================= ================================================================== ================================
+node_type   extension point         metadata definition                                                list of core extensions
```

Comment:
Fill values are kind of an extension point too, even if 3.0 does not spell it out in the extensions point section right now. The fill value metadata section has this to say 
> Extensions to the spec that define new data types must also define the JSON fill value representation.

A `string` data type needs string fill values, for example.

Reply:
I tried to address this in https://github.com/zarr-developers/zarr-specs/pull/330/commits/a791ae536d57526de4c45cf7b6e6b943e601cbd0.

Reply:
Nailed it